master/internal/rm/kubernetesrm/jobs.go

```diff
@@ -2166,6 +2180,14 @@ func (j *jobsService) verifyNamespaceExists(namespaceName string) error {
 		worker.configMapInterfaces = j.configMapInterfaces
 		worker.jobInterface = j.jobInterfaces
 	}
+
+	// Since we don't know whether the namespace exists yet, and we don't want to do duplicate
+	// namespace informers, add all non-auto-created namespaces bound to a workspace with a map.
+	if _, ok := j.namespacesWithInformers[namespaceName]; !ok {
```

Comment:
What makes things like `podInterfaces` and `jobInterfaces` a bit unreliable here is when we delete a workspace, the `podInterfaces`, `jobInterfaces`, and `configMapInterface` maps are updated to remove that namespace from their respective maps. 
So if a new workspace binds to the same namespace, it'd be "absent" from the maps but would still have informers running (since we don't cleanup the go routines). 
So we reference this map instead. I made a small adjustment so that we add to the map in `syncNamespaces` rather than just in this if-block to make sure that's guaranteed to contain any namespace that has associated informers. 

Another alternative to adding this map would just be to not update the three aforementioned maps when we delete a workspace, and then we could use those as the "source of truth" for whether or not we need to add informers for an untracked namespace.

For now assuming that we keep the map, I'll update it to be a set.Set